rustls/src/client/hs.rs

```diff
@@ -1181,6 +1179,25 @@ impl State<ClientConnectionData> for ExpectServerHelloOrHelloRetryRequest {
     }
 }
 
+fn check_certificate_type(
```

Comment:
Suggest this takes a `requires_raw_keys: bool` instead of `client_extpects: CertificateType` and a reference to `common` so that it can take care of the `send_fatal_alert()` which is currently duplicated in both callers.